apps/explorer-next/src/app/services/soul.service.ts

```diff
@@ -26,22 +32,34 @@ export class SoulService {
       );
   }
 
-  getSoulDetailsById(soulId: string) {
+  getSoulDataById(
+    soulId: string,
+    query: DocumentNode
+  ): Observable<Partial<Soul>> {
     return this.apollo
-      .watchQuery<SNIDetail>({
-        query: SOUL_DETAIL,
+      .watchQuery<SNIData>({
+        query,
         variables: {
           sniId: soulId,
         },
       })
       .valueChanges.pipe(
         map((val) => val?.data?.sni),
-        map(({ __typename, ...val }) => val),
+        map((data) => this.removeTypeName(data)),
         shareReplay()
       );
   }
 
-  setFilterToSouls(data: any) {
+  setFilterToSouls(data: SoulFilter) {
     this.subject.next(data);
   }
+
+  removeTypeName(data: Soul) {
+    return JSON.parse(
+      JSON.stringify(data, (key, value) => {
+        if (key === '__typename') return;
+        return value;
+      })
+    );
+  }
```

Comment:
I'm wondering how and why we are getting '__typename' in query so we should remove it later.